File: leetcode_ly/c++/BinarySearch/Sqrtx.cpp
class Solution {
public:
    int mySqrt(int x) {
        int low = 1;
        int high = x;
        while(low <= high){
            long middle = low + (high - low) / 2;
            if(middle * middle == x) return middle;
            else if(middle * middle > x) high = middle - 1;
            else low = middle + 1;
        }
        return high;
    }
};

File: leetcode_ly/java/Binary Search/Sqrtx.java
public class Solution {
    public int mySqrt(int x) {
        if(x == 0) return 0;
        int i = 1;
        int j = x;
        while(i <= j){
            int m = i + (j - i) / 2;
            if((long)m * m == x) return m;
            else if((long)(m + 1) * (m + 1) == x) return m + 1;
            else if((long)m * m < x && (long)(m + 1) * (m + 1) > x) return m;
            else if((long)(m + 1) * (m + 1) < x) i = m + 1;
            else j = m - 1;
        }
        return -1;
    }
}